diff -Nur -x '*.orig' -x '*~' gnome-panel-2.22.0/gnome-panel/panel-default-setup.entries gnome-panel-2.22.0.new/gnome-panel/panel-default-setup.entries
--- gnome-panel-2.22.0/gnome-panel/panel-default-setup.entries	2008-03-10 21:44:57.000000000 +0100
+++ gnome-panel-2.22.0.new/gnome-panel/panel-default-setup.entries	2008-03-14 14:42:23.000000000 +0100
@@ -35,6 +35,12 @@
           <value>
             <string>email_launcher</string>
           </value>
+          <value>
+            <string>yelp_launcher</string>
+          </value>
+          <value>
+            <string>session_dialog</string>
+          </value>
         </list>
       </value>
     </entry>
@@ -47,9 +53,6 @@
       <value>
         <list type="string">
           <value>
-            <string>window_menu</string>
-          </value>
-          <value>
             <string>mixer</string>
           </value>
           <value>
@@ -67,6 +70,12 @@
           <value>
             <string>workspace_switcher</string>
           </value>
+          <value>
+            <string>trashapplet</string>
+          </value>
+          <value>
+            <string>fast_user_switch</string>
+          </value>
         </list>
       </value>
     </entry>
@@ -262,7 +271,7 @@
       <key>objects/browser_launcher/launcher_location</key>
       <schema_key>/schemas/apps/panel/objects/launcher_location</schema_key>
       <value>
-        <string>epiphany.desktop</string>
+        <string>firefox.desktop</string>
       </value>
     </entry>
 
@@ -316,7 +325,7 @@
       <key>objects/email_launcher/launcher_location</key>
       <schema_key>/schemas/apps/panel/objects/launcher_location</schema_key>
       <value>
-        <string>evolution.desktop</string>
+        <string>evolution-mail.desktop</string>
       </value>
     </entry>
 
@@ -329,61 +338,61 @@
     <entry><key>objects/email_launcher/menu_path</key><schema_key>/schemas/apps/panel/objects/menu_path</schema_key></entry>
     <entry><key>objects/email_launcher/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
 
-  <!-- Window Menu Applet -->
+ <!-- Yelp Launcher -->
 
     <entry>
-      <key>applets/window_menu/object_type</key>
+      <key>objects/yelp_launcher/object_type</key>
       <schema_key>/schemas/apps/panel/objects/object_type</schema_key>
       <value>
-        <string>bonobo-applet</string>
+        <string>launcher-object</string>
       </value>
     </entry>
     <entry>
-      <key>applets/window_menu/toplevel_id</key>
+      <key>objects/yelp_launcher/toplevel_id</key>
       <schema_key>/schemas/apps/panel/objects/toplevel_id</schema_key>
       <value>
         <string>top_panel</string>
       </value>
     </entry>
     <entry>
-      <key>applets/window_menu/position</key>
+      <key>objects/yelp_launcher/position</key>
       <schema_key>/schemas/apps/panel/objects/position</schema_key>
       <value>
-        <int>0</int>
+        <int>4</int>
       </value>
     </entry>
     <entry>
-      <key>applets/window_menu/panel_right_stick</key>
+      <key>objects/yelp_launcher/panel_right_stick</key>
       <schema_key>/schemas/apps/panel/objects/panel_right_stick</schema_key>
       <value>
-        <bool>true</bool>
+        <bool>false</bool>
       </value>
     </entry>
     <entry>
-      <key>applets/window_menu/locked</key>
+      <key>objects/yelp_launcher/locked</key>
       <schema_key>/schemas/apps/panel/objects/locked</schema_key>
       <value>
         <bool>true</bool>
       </value>
     </entry>
     <entry>
-      <key>applets/window_menu/bonobo_iid</key>
-      <schema_key>/schemas/apps/panel/objects/bonobo_iid</schema_key>
+      <key>objects/yelp_launcher/launcher_location</key>
+      <schema_key>/schemas/apps/panel/objects/launcher_location</schema_key>
       <value>
-        <string>OAFIID:GNOME_WindowMenuApplet</string>
+        <string>file:///usr/share/applications/yelp.desktop</string>
       </value>
     </entry>
 
-    <entry><key>applets/window_menu/attached_toplevel_id</key><schema_key>/schemas/apps/panel/objects/attached_toplevel_id</schema_key> </entry>
-    <entry><key>applets/window_menu/tooltip</key><schema_key>/schemas/apps/panel/objects/tooltip</schema_key></entry>
-    <entry><key>applets/window_menu/use_custom_icon</key><schema_key>/schemas/apps/panel/objects/use_custom_icon</schema_key></entry>
-    <entry><key>applets/window_menu/custom_icon</key><schema_key>/schemas/apps/panel/objects/custom_icon</schema_key></entry>
-    <entry><key>applets/window_menu/use_menu_path</key><schema_key>/schemas/apps/panel/objects/use_menu_path</schema_key></entry>
-    <entry><key>applets/window_menu/menu_path</key><schema_key>/schemas/apps/panel/objects/menu_path</schema_key></entry>
-    <entry><key>applets/window_menu/launcher_location</key><schema_key>/schemas/apps/panel/objects/launcher_location</schema_key></entry>
-    <entry><key>applets/window_menu/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
+    <entry><key>objects/yelp_launcher/bonobo_iid</key><schema_key>/schemas/apps/panel/objects/bonobo_iid</schema_key></entry>
+    <entry><key>objects/yelp_launcher/attached_toplevel_id</key><schema_key>/schemas/apps/panel/objects/attached_toplevel_id</schema_key> </entry>
+    <entry><key>objects/yelp_launcher/tooltip</key><schema_key>/schemas/apps/panel/objects/tooltip</schema_key></entry>
+    <entry><key>objects/yelp_launcher/use_custom_icon</key><schema_key>/schemas/apps/panel/objects/use_custom_icon</schema_key></entry>
+    <entry><key>objects/yelp_launcher/custom_icon</key><schema_key>/schemas/apps/panel/objects/custom_icon</schema_key></entry>
+    <entry><key>objects/yelp_launcher/use_menu_path</key><schema_key>/schemas/apps/panel/objects/use_menu_path</schema_key></entry>
+    <entry><key>objects/yelp_launcher/menu_path</key><schema_key>/schemas/apps/panel/objects/menu_path</schema_key></entry>
+    <entry><key>objects/yelp_launcher/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
 
-  <!-- Mixer Applet -->
+ <!-- Mixer Applet -->
 
     <entry>
       <key>applets/mixer/object_type</key>
@@ -403,7 +412,7 @@
       <key>applets/mixer/position</key>
       <schema_key>/schemas/apps/panel/objects/position</schema_key>
       <value>
-        <int>1</int>
+        <int>2</int>
       </value>
     </entry>
     <entry>
@@ -457,7 +466,7 @@
       <key>applets/clock/position</key>
       <schema_key>/schemas/apps/panel/objects/position</schema_key>
       <value>
-        <int>2</int>
+        <int>1</int>
       </value>
     </entry>
     <entry>
@@ -545,6 +554,60 @@
     <entry><key>applets/notification_area/launcher_location</key><schema_key>/schemas/apps/panel/objects/launcher_location</schema_key></entry>
     <entry><key>applets/notification_area/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
 
+  <!-- Fast User Switch Applet -->
+
+    <entry>
+      <key>applets/fast_user_switch/object_type</key>
+      <schema_key>/schemas/apps/panel/objects/object_type</schema_key>
+      <value>
+        <string>bonobo-applet</string>
+      </value>
+    </entry>
+    <entry>
+      <key>applets/fast_user_switch/toplevel_id</key>
+      <schema_key>/schemas/apps/panel/objects/toplevel_id</schema_key>
+      <value>
+        <string>top_panel</string>
+      </value>
+    </entry>
+    <entry>
+      <key>applets/fast_user_switch/position</key>
+      <schema_key>/schemas/apps/panel/objects/position</schema_key>
+      <value>
+        <int>5</int>
+      </value>
+    </entry>
+    <entry>
+      <key>applets/fast_user_switch/panel_right_stick</key>
+      <schema_key>/schemas/apps/panel/objects/panel_right_stick</schema_key>
+      <value>
+        <bool>true</bool>
+      </value>
+    </entry>
+    <entry>
+      <key>applets/fast_user_switch/locked</key>
+      <schema_key>/schemas/apps/panel/objects/locked</schema_key>
+      <value>
+        <bool>true</bool>
+      </value>
+    </entry>
+    <entry>
+      <key>applets/fast_user_switch/bonobo_iid</key>
+      <schema_key>/schemas/apps/panel/objects/bonobo_iid</schema_key>
+      <value>
+        <string>OAFIID:GNOME_FastUserSwitchApplet</string>
+      </value>
+    </entry>
+
+    <entry><key>applets/fast_user_switch/attached_toplevel_id</key><schema_key>/schemas/apps/panel/objects/attached_toplevel_id</schema_key> </entry>
+    <entry><key>applets/fast_user_switch/tooltip</key><schema_key>/schemas/apps/panel/objects/tooltip</schema_key></entry>
+    <entry><key>applets/fast_user_switch/use_custom_icon</key><schema_key>/schemas/apps/panel/objects/use_custom_icon</schema_key></entry>
+    <entry><key>applets/fast_user_switch/custom_icon</key><schema_key>/schemas/apps/panel/objects/custom_icon</schema_key></entry>
+    <entry><key>applets/fast_user_switch/use_menu_path</key><schema_key>/schemas/apps/panel/objects/use_menu_path</schema_key></entry>
+    <entry><key>applets/fast_user_switch/menu_path</key><schema_key>/schemas/apps/panel/objects/menu_path</schema_key></entry>
+    <entry><key>applets/fast_user_switch/launcher_location</key><schema_key>/schemas/apps/panel/objects/launcher_location</schema_key></entry>
+    <entry><key>applets/fast_user_switch/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
+
   <!-- Show Desktop Applet -->
 
     <entry>
@@ -599,6 +662,67 @@
     <entry><key>applets/show_desktop_button/launcher_location</key><schema_key>/schemas/apps/panel/objects/launcher_location</schema_key></entry>
     <entry><key>applets/show_desktop_button/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
 
+  <!-- Session Dialog Applet -->
+
+    <entry>
+      <key>objects/session_dialog/object_type</key>
+      <schema_key>/schemas/apps/panel/objects/object_type</schema_key>
+      <value>
+        <string>action-applet</string>
+      </value>
+    </entry>
+    <entry>
+      <key>objects/session_dialog/toplevel_id</key>
+      <schema_key>/schemas/apps/panel/objects/toplevel_id</schema_key>
+      <value>
+        <string>top_panel</string>
+      </value>
+    </entry>
+    <entry>
+      <key>objects/session_dialog/position</key>
+      <schema_key>/schemas/apps/panel/objects/position</schema_key>
+      <value>
+        <int>0</int>
+      </value>
+    </entry>
+    <entry>
+      <key>objects/session_dialog/panel_right_stick</key>
+      <schema_key>/schemas/apps/panel/objects/panel_right_stick</schema_key>
+      <value>
+        <bool>true</bool>
+      </value>
+    </entry>
+    <entry>
+      <key>objects/session_dialog/locked</key>
+      <schema_key>/schemas/apps/panel/objects/locked</schema_key>
+      <value>
+        <bool>true</bool>
+      </value>
+    </entry>
+    <entry>
+      <key>objects/session_dialog/action_type</key>
+      <schema_key>/schemas/apps/panel/objects/action_type</schema_key>
+      <value>
+        <string>logout</string>
+      </value>
+    </entry>
+    <entry>
+      <key>objects/session_dialog/launcher_location</key>
+      <schema_key>/schemas/apps/panel/objects/launcher_location</schema_key>
+      <value>
+        <string></string>
+      </value>
+    </entry>
+ 
+    <entry><key>objects/session_dialog/bonobo_iid</key><schema_key>/schemas/apps/panel/objects/bonobo_iid</schema_key></entry>
+    <entry><key>objects/session_dialog/attached_toplevel_id</key><schema_key>/schemas/apps/panel/objects/attached_toplevel_id</schema_key> </entry>
+    <entry><key>objects/session_dialog/tooltip</key><schema_key>/schemas/apps/panel/objects/tooltip</schema_key></entry>
+    <entry><key>objects/session_dialog/use_custom_icon</key><schema_key>/schemas/apps/panel/objects/use_custom_icon</schema_key></entry>
+    <entry><key>objects/session_dialog/custom_icon</key><schema_key>/schemas/apps/panel/objects/custom_icon</schema_key></entry>
+    <entry><key>objects/session_dialog/use_menu_path</key><schema_key>/schemas/apps/panel/objects/use_menu_path</schema_key></entry>
+    <entry><key>objects/session_dialog/menu_path</key><schema_key>/schemas/apps/panel/objects/menu_path</schema_key></entry>
+    <entry><key>objects/session_dialog/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
+
   <!-- Window List Applet -->
 
     <entry>
@@ -673,7 +797,7 @@
       <key>applets/workspace_switcher/position</key>
       <schema_key>/schemas/apps/panel/objects/position</schema_key>
       <value>
-        <int>0</int>
+        <int>1</int>
       </value>
     </entry>
     <entry>
@@ -707,6 +831,59 @@
     <entry><key>applets/workspace_switcher/launcher_location</key><schema_key>/schemas/apps/panel/objects/launcher_location</schema_key></entry>
     <entry><key>applets/workspace_switcher/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
 
+  <!-- TrashApplet Applet -->
+
+    <entry>
+      <key>applets/trashapplet/object_type</key>
+      <schema_key>/schemas/apps/panel/objects/object_type</schema_key>
+      <value>
+        <string>bonobo-applet</string>
+      </value>
+    </entry>
+    <entry>
+      <key>applets/trashapplet/toplevel_id</key>
+      <schema_key>/schemas/apps/panel/objects/toplevel_id</schema_key>
+      <value>
+        <string>bottom_panel</string>
+      </value>
+    </entry>
+    <entry>
+      <key>applets/trashapplet/position</key>
+      <schema_key>/schemas/apps/panel/objects/position</schema_key>
+      <value>
+        <int>0</int>
+      </value>
+    </entry>
+    <entry>
+      <key>applets/trashapplet/panel_right_stick</key>
+      <schema_key>/schemas/apps/panel/objects/panel_right_stick</schema_key>
+      <value>
+        <bool>true</bool>
+      </value>
+    </entry>
+    <entry>
+      <key>applets/trashapplet/locked</key>
+      <schema_key>/schemas/apps/panel/objects/locked</schema_key>
+      <value>
+        <bool>true</bool>
+      </value>
+    </entry>
+    <entry>
+      <key>applets/trashapplet/bonobo_iid</key>
+      <schema_key>/schemas/apps/panel/objects/bonobo_iid</schema_key>
+      <value>
+        <string>OAFIID:GNOME_Panel_TrashApplet</string>
+      </value>
+    </entry>
+
+    <entry><key>applets/trashapplet/attached_toplevel_id</key><schema_key>/schemas/apps/panel/objects/attached_toplevel_id</schema_key> </entry>
+    <entry><key>applets/trashapplet/tooltip</key><schema_key>/schemas/apps/panel/objects/tooltip</schema_key></entry>
+    <entry><key>applets/trashapplet/use_custom_icon</key><schema_key>/schemas/apps/panel/objects/use_custom_icon</schema_key></entry>
+    <entry><key>applets/trashapplet/custom_icon</key><schema_key>/schemas/apps/panel/objects/custom_icon</schema_key></entry>
+    <entry><key>applets/trashapplet/use_menu_path</key><schema_key>/schemas/apps/panel/objects/use_menu_path</schema_key></entry>
+    <entry><key>applets/trashapplet/menu_path</key><schema_key>/schemas/apps/panel/objects/menu_path</schema_key></entry>
+    <entry><key>applets/trashapplet/launcher_location</key><schema_key>/schemas/apps/panel/objects/launcher_location</schema_key></entry>
+
   </entrylist>
 
 </gconfentryfile>
