--- gnome-panel/panel-menu-items.c	2007-02-27 05:41:11.000000000 +0100
+++ gnome-panel/panel-menu-items.c.new	2007-02-27 05:50:50.000000000 +0100
@@ -745,35 +745,26 @@ panel_desktop_menu_item_append_menu (Gtk
 	add_menu = create_empty_menu ();
 	gtk_menu_item_set_submenu (GTK_MENU_ITEM (item), add_menu);
 	
-	panel_menu_items_append_from_desktop (add_menu, "yelp.desktop", _("System Documentation"));
+	panel_menu_items_append_from_desktop (add_menu, "guadalinex-user-manual.desktop", _("Guía de usuario de Guadalinex"));
 	
 	panel_menu_items_append_place_item ("gnome-globe",
-					    _("Online Documentation"),
-					    _("Find more documentation online"),
+					    _("Asistencia (por Internet)"),
+					    _("Obtener ayuda a través de Internet"),
 					    add_menu,
 					    G_CALLBACK (activate_uri),
-					    "http://help.ubuntu.com");
+					    "http://www.guadalinex.org/distro/V4/ayuda/");
 	
 	panel_menu_items_append_place_item ("stock_new-meeting",
-					    _("Community Support"),
-					    _("Get help from other Ubuntu users"),
+					    _("Comunidad Guadalinex"),
+					    _("Obtener ayuda de otros usuarios de Guadalinex"),
 					    add_menu,
 					    G_CALLBACK (activate_uri),
-					    "http://www.ubuntu.com/support/supportoptions/freesupport");
+					    "http://www.guadalinex.org/distro/V4/foros/");
 
-	panel_menu_items_append_place_item ("stock_send-fax",
-					    _("Commercial Support"),
-					    _("Find a commercial support provider"),
-					    add_menu,
-					    G_CALLBACK (activate_uri),
-					    "http://www.ubuntu.com/support/marketplace");
 
-	panel_menu_items_append_place_item ("stock_book_open",
-					    _("Ubuntu Book Excerpt"),
-					    _("Read selected book chapters"),
-					    add_menu,
-					    G_CALLBACK (activate_uri),
-					    "http://help.ubuntu.com/6.10/book/book-toc.html");
+	panel_menu_items_append_from_desktop (add_menu, "hermesdruid.desktop", NULL);
+
+	panel_menu_items_append_from_desktop (add_menu, "guadalinex-about.desktop", NULL);
 
 	panel_menu_items_append_from_desktop (menu, "gnome-about.desktop", NULL);
 
